x-pack/plugins/transform/public/app/sections/transform_management/components/edit_transform_flyout/use_edit_transform_flyout.ts

```diff
@@ -107,15 +108,40 @@ type Validator = (value: any, isOptional?: boolean) => string[];
 // We do this so we have fine grained control over field validation and the option to
 // cast to special values like `null` for disabling `docs_per_second`.
 const numberAboveZeroNotValidErrorMessage = i18n.translate(
-  'xpack.transform.transformList.editFlyoutFormNumberNotValidErrorMessage',
+  'xpack.transform.transformList.editFlyoutFormNumberAboveZeroNotValidErrorMessage',
   {
     defaultMessage: 'Value needs to be an integer above zero.',
   }
 );
-export const integerAboveZeroValidator: Validator = (value) =>
-  !isNaN(value) && Number.isInteger(+value) && +value > 0 && !(value + '').includes('.')
-    ? []
-    : [numberAboveZeroNotValidErrorMessage];
+
+const numberGreaterThanOrEqualToNegativeOneNotValidErrorMessage = i18n.translate(
+  'xpack.transform.transformList.editFlyoutFormNumberGreaterThanOrEqualToNegativeOneNotValidErrorMessage',
+  {
+    defaultMessage: 'Number of retries needs to be a number greater than or equal to -1.',
+  }
+);
+
+export const integerGreaterThanOrEqualToValidatorFactory =
```

Comment:
can we move these [validators](https://github.com/elastic/kibana/blob/6ff66841a7ddb85d7c15e872b39b872dc9eee378/x-pack/plugins/ml/common/util/validators.ts#L110) to the package and use it here instead?

Reply:
If we use validators, does that mean you can use the same error message in both the wizard and the edit form? Currently a different message is used in each place.

Reply:
Since the validator needs to check for < 100, I ended up removing this. We can refactor out the validators in a follow up PR as it is out of scope of this PR.

Reply:
As I mentioned in the first comment, worth moving the [number validator](https://github.com/elastic/kibana/blob/6ff66841a7ddb85d7c15e872b39b872dc9eee378/x-pack/plugins/ml/common/util/validators.ts#L110) to the recently created ml-utils package